src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -2079,6 +2079,16 @@ int fmi3_xml_handle_ModelVariables(fmi3_xml_parser_context_t *context, const cha
                 jm_vector_set_item(jm_voidp)(md->variablesByVR, i, jm_vector_get_item(jm_named_ptr)(&md->variablesByName, i).ptr);
             }
             jm_vector_qsort(jm_voidp)(md->variablesByVR, fmi3_xml_compare_vr_and_original_index);
+
+            for (size_t i = 0; i < size-1; ++i) {
+                fmi3_xml_variable_t* v1 = jm_vector_get_item(jm_voidp)(md->variablesByVR, i);
+                fmi3_xml_variable_t* v2 = jm_vector_get_item(jm_voidp)(md->variablesByVR, i+1);
+                if (v1->vr == v2->vr) {
+                    fmi3_xml_parse_error(context, "The following variables have the same value reference: %s, %s",
+                            v1->name, v2->name);
+                    // Usually we return -1 on error, but should be fine to continue.
```

Comment:
@PeterMeisrimelModelon
It's not really clear how we should handle parse errors (in general). In this case I thought it overly dramatic to stop the whole ModelVariable handler, because that will leave everything broken. I know other code in same function do it, but unclear why.

Reply:
Partially. The problem is if we give a fatal error then we don't return anything from the parse call. I.e. returned fmi3_import_t* will be NULL. So let's say you just wanted to display the parsed variables in a GUI, then you can't even do that.
However, it may lead to secondary errors when parsing ModelStructure, or ambiguities when trying to retrieve the variable with that VR.

So it's not clear what is best, and we don't have any clear guidelines of when something should be fatal. But I could go with parse_fatal in this case. It's easier to relax that restriction than going the other way.

Reply:
So my proposal is to change to parse_fatal, would that be OK?

Reply:
If I understand this correctly, fmi3_xml_parse_fatal would stop parsing, give and appropriate log message, but you would still get the parsed result until that point?

Reply:
You only get the log.

`fmi3_xml_parse_fatal` calls `XML_StopParser(context->parser,0);`, where arg2==0 which means "won't resume".

`fmi3_import_parse_xml` calls `fmi3_xml_parse_model_description` calls `XML_Parse`. All our handlers are invoked from that function. Inside that function we call parse_fatal, which leads to `XML_Parse` returning some ERROR enum, because we called StopParser with "won't resume", which in turn leads to us eventually returning NULL for the whole parse call.

Reply:
Ah, thanks for clarifying. Sounds good to me.